library/ssl_srv.c

```diff
@@ -3830,7 +3830,7 @@ static int ssl_parse_client_psk_identity( mbedtls_ssl_context *ssl, unsigned cha
     n = ( (*p)[0] << 8 ) | (*p)[1];
     *p += 2;
 
-    if( n < 1 || n > 65535 || n > (size_t) ( end - *p ) )
+    if( !n || n > (uint16_t) ( end - *p ) )
```

Comment:
Please don't use `!n` when `n` isn't a boolean. `n < 1` is fine here, even now that `n` is unsigned.

Reply:
I suggest to change `!n` to `n == 0`.